lib/modware/stack.rb

```diff
@@ -1,11 +1,16 @@
-require 'key_struct'
+# frozen_string_literal: true
 
 module Modware
   class Stack
     def initialize(env:)
       @env_klass = case env
                    when Class then env
-                   else KeyStruct[*env]
+                   else
+                     Struct.new(*env, keyword_init: true) do
+                       def to_hash
+                         to_h.compact
```

Comment:
@urkle using `compact` here is affecting the expectations of nil values https://github.com/SchemaPlus/schema_plus_core/runs/6015921381?check_suite_focus=true#step:9:19